include/deal.II/base/config.h.in

```diff
@@ -545,6 +545,19 @@ _Pragma("GCC diagnostic pop")
   #define DEAL_II_WARNING(desc) DEAL_II_DO_PRAGMA(message(#desc))
 #endif
 
+/***********************************************************************
+ * Some places in deal.II require or benefit from using "if constexpr".
+ * However, this feature is only available in C++17 and above.
+ * The macro defined here expands to "if constexpr(...)" when C++17 is supported
+ * and to "if (...)" otherwise.
+ */
+#ifdef DEAL_II_HAVE_CXX17
+  #define DEAL_II_IF_CONSTEXPR(x) if constexpr x
+#else
+  #define DEAL_II_IF_CONSTEXPR(x) if x
+#endif
```

Comment:
What's the advantage of `if constexpr x` over `if constexpr (x)`?

Reply:
For anything including a comma we need to use double parentheses anyway and this forces double parentheses for every condition. I thought that would be good for consistency but I'm happy to change.